runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -145,7 +145,13 @@ MM_IndexableObjectAllocationModel::initializeIndexableObject(MM_EnvironmentBase
 	/* Lay out arraylet and arrayoid pointers */
 	switch (_layout) {
 	case GC_ArrayletObjectModel::InlineContiguous:
-		Assert_MM_true(1 == _numberOfArraylets);
+		/**
+		* We now treat 0-stride element inside flattened array as inlineContiguous instead of discontiguous
+		* And in this situation, the _numberOfArraylets will be 0 instead of 1
+		* Note the _dataSize can be 0 only because of 0-stride,
+		* since 0-length array is treated as discontiguous
+		*/
```

Comment:
I am trying to reword this comment with accent why _numberOfArraylets can be not equal 1 for InlineContiguous case.
Something like this:

_numberOfArraylets is set to 0 for 0-stride Flattened array. This case can be recognized by checking _dataSize is 0.
Please note there are two cases when _dataSize is set to 0: 0-stride Flattened array and 0-length array 
but latter one is treated as Discontiguous out of this path.

Reply:
Should we also mention that we switch to inlineContiguous here since we used discontiguous before?

Reply:
No, I think there is no "before". 0-stride elements Flattened arrays were not supported previously.